container/runtime/src/main/java/org/wildfly/swarm/container/runtime/RuntimeDeployer.java

```diff
@@ -69,13 +71,31 @@ public void debug(boolean debug) {
     @Override
     public void deploy(Archive<?> deployment) throws DeploymentException {
 
+        // 1. create a meta data index
+        Indexer indexer = new Indexer();
+        Map<ArchivePath, Node> c = deployment.getContent();
+        try {
+            for (Map.Entry<ArchivePath, Node> each : c.entrySet()) {
+                if(each.getKey().get().endsWith(".class")) {
+                    indexer.index(each.getValue().getAsset().openStream());
+                }
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+
+        Index index = indexer.complete();
+
+
+        // 2. give the factions a chance to handle the deployment
         for (ServerConfiguration each : this.configurations) {
             each.prepareArchive(deployment);
+            each.processMetaData(deployment, index);
```

Comment:
We may want to do the metadata processing in a loop after this one, to account for one Fraction doing something in prepareArchive() that impacts another Fractions processMetaData().